src/preload.js

```diff
@@ -243,26 +249,63 @@ function updateMediaInfo(options, notify) {
  */
 setInterval(function () {
   const title = elements.getText("title");
-  const url = elements.get("url").href.replace(/[^0-9]/g, "");
   const artists = elements.getText("artists");
+  const current = elements.getText("current");
+  const duration = elements.getText("duration");
+  const barval = elements.get("bar").getAttribute("aria-valuenow");
   const songDashArtistTitle = `${title} - ${artists}`;
   const currentStatus = getCurrentlyPlayingStatus();
   const options = {
     title,
     message: artists,
     status: currentStatus,
-    url: `https://tidal.com/browse/track/${url}`,
+    url: currentURL,
+    current: current,
+    duration: duration,
   };
 
   const playStatusChanged = currentStatus !== currentPlayStatus;
+  const barvalChanged = barval !== barvalue;
   const titleOrArtistChanged = currentSong !== songDashArtistTitle;
 
-  if (titleOrArtistChanged || playStatusChanged) {
+  if (titleOrArtistChanged || playStatusChanged || barvalChanged || updatecurrent) {
```

Comment:
This means you'll send a status update every second or so because the bar changes every second.
Won't that trip the rate limiter?

Reply:
the bar value that I get from the element actually doesnt change any second it only changes on play, pause and when you skip to a certain part of the song

Reply:
I don't get it. The evaluation of currentTimeChanged is done on [line 285](https://github.com/Mastermindzh/tidal-hifi/pull/43/files#diff-032e180739c4103bc3c3e71de4e372ef41482cfedaf727481a9f0724a5763f5bR285).

You check whether the current bar time (which I presume would update every second as it shows seconds in the UI) is different from "progressBarTime".

Whenever that is true, a message is being sent, regardless of the other code. (the promise will always be initiated and thus resolved, triggering an update.


<h4>additionally</h4>

progressBarTime is set on both lines [297](https://github.com/Mastermindzh/tidal-hifi/pull/43/files#diff-032e180739c4103bc3c3e71de4e372ef41482cfedaf727481a9f0724a5763f5bR297) and [313](https://github.com/Mastermindzh/tidal-hifi/pull/43/files#diff-032e180739c4103bc3c3e71de4e372ef41482cfedaf727481a9f0724a5763f5bR313) with if statements that are directly opposed to each other. (because the additional check on line 297 is not needed, you already do that on 271)

So, no matter what the entrypoint is one of the ifs will always trigger and the global progressBarTime will be updated.

Reply:
Or are you saying "*[data-test="progress-bar"]" doesn't update itself every second?
I can test this out tonight to verify as well. Might be able to wrap it up today.

Reply:
> Or are you saying "*[data-test="progress-bar"]" doesn't update itself every second?
> I can test this out tonight to verify as well. Might be able to wrap it up today.

the property aria-valuenow which line 272 gets is part of the HTML element where the data-test is located that property only changes like I said on play, pause and when someone skips parts of the song.

> the additional check on line 297 is not needed, you already do that on 271

The check in line 297 exists cause if you use the skip or reverse buttons(in the player or if you have media keys also with the media keys) the current time tends to break as Tidal does some weird stuff with the current text so this check simply makes sure that the current time actually is correct.

> (which I presume would update every second as it shows seconds in the UI) is different from "progressBarTime"

Funnily enough the UI only updates an span which has the time via JS until you pause which then makes it update in the element like I mentioned above in the first section.

> progressBarTime is set on both lines 297 and 313 with if statements that are directly opposed to each other. 

I can totally understand the confusion but that part is to blame on Tidal's system of them handling the time. even if line 313 sets progressBarTime it still ends up changing around almost 1 second later on Tidal's site which the check on 297 actually notices and resets it to the proper time.

Reply:
Here are two gifs showing it in action:
![gif 1](https://i.imgur.com/tZoqYQt.gif)
![gif 2](https://i.imgur.com/2Jie3HL.gif)

Reply:
This is what happens if you remove currentTimeChanged and you start skipping songs or going back it starts to desync 
![image](https://user-images.githubusercontent.com/8841466/115708945-e8109e80-a370-11eb-91c3-88180844499d.png)

Reply:
Ok, great, thanks for looking it all up for me 😄 
I'll have a look tonight!

Reply:
Alright, Imma just continue using it until you check it out yourself.